src/components/side_panel/pivot/pivot_side_panel/pivot_side_panel_store.ts

```diff
@@ -215,13 +217,18 @@ export class PivotSidePanelStore extends SpreadsheetStore {
     const { columns, rows } = definition;
     const columnsWithGranularity = deepCopy(columns);
     const rowsWithGranularity = deepCopy(rows);
-    const unusedGranularities = this.getUnusedDateTimeGranularities(fields, definition);
+    const unusedGranularities = this.getUnusedDateTimeGranularities(fields, definition, undefined);
     for (const dimension of columnsWithGranularity.concat(rowsWithGranularity)) {
       const fieldType = fields[dimension.name]?.type;
       if ((fieldType === "date" || fieldType === "datetime") && !dimension.granularity) {
-        const granularity = unusedGranularities[dimension.name]?.values().next().value || "year";
```

Comment:
I'd keep the previous simpler code. We don't care if month or year is the first default value through the side panel UI. No need for a special case